pkg/authorization/workspace_content_authorizer.go

```diff
@@ -92,6 +92,22 @@ func (a *workspaceContentAuthorizer) Authorize(ctx context.Context, attr authori
 	isServiceAccountFromRootCluster := subjectClusters[tenancyv1alpha1.RootCluster]
 	isServiceAccountFromCluster := subjectClusters[cluster.Name]
 
+	if IsDeepSubjectAccessReviewFrom(ctx, attr) {
+		attr := deepCopyAttributes(attr)
+		// this is a deep SAR request, we have to skip the checks here and delegate to the subsequent authorizer.
+		if isAuthenticated && !isUser && !isServiceAccountFromCluster {
+			// anonymize service accounts from other workspaces
+			attr.User = &user.DefaultInfo{
+				Name:   "system:anonymous",
+				Groups: []string{"system:authenticated"},
+				Extra: map[string][]string{
+					authserviceaccount.ClusterNameKey: {cluster.Name.String()},
```

Comment:
@sttts i believe this is correct. The anonymous service account must be bound to the requested cluster (see added unit test https://github.com/kcp-dev/kcp/pull/1739/files#diff-7b5431a2819884e52f3d2ab49ef1c44202fb3c3149873b1be4566ace41258fecR208)

Reply:
not sure I understand. We anonymize the request. I.e. technically the user is no service account anymore in the remaining authz.

Reply:
true, i'll remove the cluster name key one